sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/CosmosAsyncItemResponse.java

```diff
@@ -4,40 +4,41 @@
 
 import com.azure.cosmos.implementation.Document;
 import com.azure.cosmos.implementation.ResourceResponse;
+import com.azure.cosmos.implementation.Utils;
 import org.apache.commons.lang3.StringUtils;
 
-public class CosmosAsyncItemResponse extends CosmosResponse<CosmosItemProperties> {
-    private final CosmosAsyncItem itemClient;
-
-    CosmosAsyncItemResponse(ResourceResponse<Document> response, PartitionKey partitionKey,
-                            CosmosAsyncContainer container) {
+public class CosmosAsyncItemResponse<T> extends CosmosResponse<CosmosItemProperties> {
+    private final Class<T> itemClassType;
+    private final String responseBodyString;
+    
+    CosmosAsyncItemResponse(ResourceResponse<Document> response, Class<T> klass) {
         super(response);
-        String bodyAsString = response.getBodyAsString();
-        if (StringUtils.isEmpty(bodyAsString)) {
+        this.itemClassType = klass;
+        responseBodyString = response.getBodyAsString();
+        if (StringUtils.isEmpty(responseBodyString)){
             super.setProperties(null);
-            itemClient = null;
         } else {
-            CosmosItemProperties props = new CosmosItemProperties(bodyAsString);
+            CosmosItemProperties props = new CosmosItemProperties(responseBodyString);
             super.setProperties(props);
-            itemClient = new CosmosAsyncItem(props.getId(), partitionKey, container);
         }
     }
 
     /**
-     * Gets the itemSettings
+     * Gets the resource .
      *
-     * @return the itemSettings
+     * @return the resource
      */
-    public CosmosItemProperties getProperties() {
-        return super.getProperties();
+    public T getResource(){
+        return Utils.parse(responseBodyString, itemClassType);
```

Comment:
Since `responseBodyString` can be empty string, can we please make sure it is handled well when parsing it to an `itemClassType` ?
Do we want to throw error in that case, or return null ?

Reply:
Handled the null/empty case. Returning null